packages/smooth_app/lib/cards/product_cards/smooth_product_card_not_found.dart

```diff
@@ -58,16 +58,14 @@ class SmoothProductCardNotFound extends StatelessWidget {
               icon: Icons.add,
               padding: const EdgeInsets.symmetric(vertical: LARGE_SPACE),
               onPressed: () async {
-                await Navigator.push<void>(
-                  context,
-                  MaterialPageRoute<void>(
-                    builder: (BuildContext context) =>
-                        AddNewProductPage(barcode),
-                  ),
+                AppNavigator.of(context).push(
+                  AppRoutes.PRODUCT_CREATOR(barcode),
                 );
-                if (callback != null) {
-                  await callback!();
-                }
+
+                // TODO(g123k): Find another way
+                // if (callback != null) {
+                //   await callback!();
+                // }
```

Comment:
`PRODUCT_CREATOR` actually opens `AddNewProductPage`.
The only problem we have now is that the `push` method doesn't return any result.

Reply:
> `PRODUCT_CREATOR` actually opens `AddNewProductPage`. The only problem we have now is that the `push` method doesn't return any result.

Then perhaps the concept of `PRODUCT_CREATOR` is flawed, or at least we should put it on the shelf.
To me the use-case is a user browsing and clicking on a link that automagically opens the app. A link that opens a known product, I can understand. A link that opens the search, I can understand. A link that opens an edit page for an unknown barcode, given that we scan on the app and we search on the app, I cannot quite picture that.
That said, perhaps the `callback` (that we actually set only once in that case) is redundant with whatever we do at the end of the `AddNewProductPage`, to be investigated.

But in general, I think that we assumed a long time ago that the deep-links were rather "read/display" pages than "write/edit" pages, specifically because they don't return values. Maybe it's not relevant, after all.

Reply:
I understand your point, but the routes in GoRouter, doesn't necessary mean they have to be opened via a deep link. In that case, `PRODUCT_CREATOR` is just a named route, as we can find in the Navigator v1.

I have handled the case where a product doesn't exist, as it may happen.
Let's say you receive a link by email and the sender has made a typo => we have to do something for that.

Reply:
> I understand your point, but the routes in GoRouter, doesn't necessary mean they have to be opened via a deep link. In that case, `PRODUCT_CREATOR` is just a named route, as we can find in the Navigator v1.

Ok, but in that case if we want to handle the callback that you commented/`TODO`ed, perhaps we would be better off making this page a full screen dialog instead of a page.

> I have handled the case where a product doesn't exist, as it may happen. Let's say you receive a link by email and the sender has made a typo => we have to do something for that.

In that case I would rather land on a "The product doesn't exist" page (with a "do you want to create it?" button).
If not we somehow ask users to create new products with wrong barcodes!

Reply:
For the first point, OK I will revert what(s done here
Mmm for the second point, I let @teolemon what's the best implementation to use.

Reply:
Mmh.. I can see some potential abuse potential but also the ability for third party apps to delegate some of their duties to Open Food Facts when a product is missing (some are currently using a web link)
I'd be in favor of giving the product addition approach a try, and kill it if we get abuse

Reply:
I don't believe we are. We can open it from scan, search, and now deep links, right ?

Reply:
Yes, exactly. Is-it useful to add it, or should I leave it as is?